network/f5/bigip_virtual_server.py

```diff
@@ -236,7 +237,7 @@ def set_rules(api, name, rules_list):
         return False
     rules_list = list(enumerate(rules_list))
     try:
-        current_rules = map(lambda x: (x['priority'], x['rule_name']), get_rules(api, name))
+        current_rules = list(map(lambda x: (x['priority'], x['rule_name']), get_rules(api, name)))
```

Comment:
As @caphrim007 mentioned, this should go in as a separate PR as it does fix a bug introduced recently in Ansible that introduces new default map() behavior.